### PR TITLE
TS-4891: Initializing prefix_str on declaration

### PR DESCRIPTION
**Pull request by persiaAziz**
prefix_str should be initialized before setup_diagslog() is called in Diags.cc:146 since a failure in setup_diagslog can lead to a call of print_va which reads from prefix_str.

Following trace might be helpful to understand the issue
https://gist.github.com/persiaAziz/48a321a1c1913af1580261fa40d14df0

### CODE REVIEW

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/861/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/757/ for details.

**Comment by persiaAziz** [comment]
Please review the changes

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/767/ for details.

**Comment by atsci** [comment]
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/871/ for details.

**Comment by persiaAziz** [comment]
Please approve again. The previous jenkins build failed because of wrong function call in Diags.i . I don't know why my compiler didn't complain about that

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/872/ for details.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/768/ for details.

**Comment by persiaAziz** [comment]
Ahh.  I didn't know about "make test", that is why I wasn't seeing these errors. Please approve again. The build should pass this time.

File: proxy/Main.cc [hunk 1]
```diff
@@ -1507,9 +1507,8 @@ main(int /* argc ATS_UNUSED */, const char **argv)
   // re-start it again, TS will crash.
   // This is also needed for log rotation - setting up the file can cause privilege
   // related errors and if diagsConfig isn't get up yet that will crash on a NULL pointer.
-  diagsConfig       = new DiagsConfig(DIAGS_LOG_FILENAME, error_tags, action_tags, false);
-  diags             = diagsConfig->diags;
-  diags->prefix_str = "Server ";
+  diagsConfig = new DiagsConfig("Server ", DIAGS_LOG_FILENAME, error_tags, action_tags, false);
```
Comment (jpeach) [inline proxy/Main.cc]: Why do we sometimes have a trailing space on the prefix?

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/874/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/770/ for details.

File: proxy/Main.cc (hunk 1 above)
Comment (SolidWallOfCode) [inline proxy/Main.cc]: I favor removing the spaces from `prefix_str` and putting it in the formatting logic. I suspect it was done this way originally to allow not having a prefix but I think we can dispose of that.

**Comment by persiaAziz** [comment]
Please review

**Comment by jpeach** [comment]
[approve ci]

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/776/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/881/ for details.

**Pull request closed without merging**